src/components/UserSelector.tsx

```diff
@@ -12,24 +24,42 @@ export const UserSelector: React.FC = () => {
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => setMenuIsOpen(!menuIsOpen)}
         >
-          <span>Choose a user</span>
+          <span>
+            {selectedUser ? selectedUser?.name : 'Choose a user'}
+          </span>
 
           <span className="icon is-small">
             <i className="fas fa-angle-down" aria-hidden="true" />
           </span>
         </button>
       </div>
 
-      <div className="dropdown-menu" id="dropdown-menu" role="menu">
-        <div className="dropdown-content">
-          <a href="#user-1" className="dropdown-item">Leanne Graham</a>
-          <a href="#user-2" className="dropdown-item is-active">Ervin Howell</a>
-          <a href="#user-3" className="dropdown-item">Clementine Bauch</a>
-          <a href="#user-4" className="dropdown-item">Patricia Lebsack</a>
-          <a href="#user-5" className="dropdown-item">Chelsey Dietrich</a>
+      {menuIsOpen ? (
+        <div className="dropdown-menu" id="dropdown-menu" role="menu">
+          <div className="dropdown-content">
+            {users.map(user => (
+              <a
+                href={`#user-${user.id}`}
+                className={classNames(
+                  'dropdown-item',
+                  {
+                    'is-active': user === selectedUser,
+                  },
+                )}
+                onClick={() => {
```

Comment:
Avoid using arrow functions in event handlers for better performance. Create a separate function for handling onClick events.